src/qibo/tomography/gate_set_tomography.py

```diff
@@ -0,0 +1,371 @@
+        invert_register (tuple): Qubit(s) to reset: Use a tuple to specify which qubit(s) to reset:
+            - (0,) to reset qubit 0;
+            - (1,) to reset qubit 1; or
+            - (0,1) to reset both qubits.
+    Returns:
+        inverse_circuit (:class:`qibo.models.Circuit`): Inverse of the input circuit's register.
+    """
+    valid_registers = [(0,), (1,), (0, 1)]
+    if invert_register is not None:
+        if (
+            not isinstance(invert_register, tuple)
+            or invert_register not in valid_registers
+        ):
+            raise_error(
+                NameError,
+                f"Invalid register {invert_register}, please pick one in {valid_registers}.",
+            )
+
+        elif len(invert_register) == 1:
+            register_to_reset = invert_register[0]
+            new_circ = Circuit(1)
+            # for data in circuit.raw["queue"]:
+            #     init_kwargs = data.get("init_kwargs", {})
+            #     if data["_target_qubits"][0] == register_to_reset:
+            # new_circ.add(getattr(gates, data["_class"])(0, **init_kwargs))
+            for gate in circuit.queue:
+                if register_to_reset in gate.target_qubits:
+                    new_circ.add(gate.__class__(0, **gate.init_kwargs))
+
+        else:
+            new_circ = circuit.copy()
+
+    return new_circ.invert()
+
+
+def _expectation_value(circuit, j, nshots=int(1e4), backend=None):
+    """Executes a circuit used in gate set tomography and processes the
+        measurement outcomes for the Pauli Transfer Matrix notation. The circuit
+        should already have noise models implemented, if any, prior to using this
+        function.
+
+        The function returns the expectation value given by either
+        :math:`\\text{tr}(M_j rho_k)` or :math:`\\Tr(M_j O_l rho_k)`,
+        where :math:`k` is the index of the state prepared (which is not necessary
+        in this function since it has been used earlier), :math:`j` is the index
+        of the measurement basis, and :math:`O_l` is the :math:`i`-th gate of
+        the circuit.
+
+    Args:
+        circuit (:class:`qibo.models.Circuit`): The Qibo circuit to be executed.
+        j (int): The index of the measurement basis.
+        nshots (int, optional): Number of shots to execute the circuit with.
+    Returns:
+        numpy.float: Expectation value.
+    """
+
+    nqubits = circuit.nqubits
+    if not nqubits in (1, 2):
+        raise_error(
+            ValueError,
+            f"nqubits given as {nqubits}. nqubits needs to be either 1 or 2.",
+        )
+
+    else:
+        if j == 0:
+            return 1.0
+        if backend is None:  # pragma: no cover
+            backend = GlobalBackend()
+
+        result = backend.execute_circuit(circuit, nshots=nshots)
+        observable = _get_observable(j, nqubits)
+        return result.expectation_from_samples(observable)
+
+
+def _gate_set_tomography(
```

Comment:
I wouldn't call it `_gate_set_tomography` as you are actually performing tomography for a single gate, maybe something like `gate_tomography` or `_gate_tomography`, depending on whether we want to expose this to the user, would be more appropriate.

Reply:
Fair enough, we can call it "gate tomography" whilst using "GST" as the wrapper since it takes in a list of gates.